### PR TITLE
Add supported platform and language versions

### PR DESCRIPTION
**Pull request by scottyab**
Similar to the Android sample [here](https://github.com/helpscout/beacon-android-sdk-sample/pull/174), we're adding Supported platform and language versions section to the readme.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,6 +11,12 @@ The Beacon SDK 2.0 requires iOS 11.0+.
 As of Beacon 2.1.0 Xcode 12 is required to build with the Beacon SDK.
 Beacon 2.0.x requires Xcode 11.4
 
+### Supported platform and language versions
+
+* iOS 11.0 to 14.0
+* Swift 5.4
```
Comment (ScottPetit) [inline README.md]:
```suggestion
* Swift 5
```

File: README.md [hunk 2]
```diff
@@ -11,6 +11,12 @@ The Beacon SDK 2.0 requires iOS 11.0+.
 As of Beacon 2.1.0 Xcode 12 is required to build with the Beacon SDK.
 Beacon 2.0.x requires Xcode 11.4
 
+### Supported platform and language versions
+
+* iOS 11.0 to 14.0
```
Comment (ScottPetit) [inline README.md]:
```suggestion
* iOS 11 to 14
```

File: README.md [hunk 3]
```diff
@@ -11,6 +11,12 @@ The Beacon SDK 2.0 requires iOS 11.0+.
...
+* Swift 5.4
+
```
Comment (ScottPetit) [inline README.md]:
```suggestion
* Xcode 12
```

**Review by ScottPetit — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (scottyab) [reply README.md]: Sorry @ScottPetit this and the other points were totally a cut and paste issue! I was listening to you when you mentioned to drop the minor version 🤦

Comment (ScottPetit) [reply README.md]: Haha all good, figured as much.

**Review by ScottPetit — APPROVED** [review]
👍

**Pull request merged**